src/Aspire.Hosting.Azure/Bicep/AzureBicepServiceBusResource.cs

```diff
@@ -39,20 +101,94 @@ public static class AzureBicepServiceBusExtensions
     /// </summary>
     /// <param name="builder">The <see cref="IDistributedApplicationBuilder"/>.</param>
     /// <param name="name">The name of the resource. This name will be used as the connection string name when referenced in a dependency.</param>
-    /// <param name="queueNames">A list of queue names associated with this service bus resource.</param>
-    /// <param name="topicNames">A list of topic names associated with this service bus resource.</param>
     /// <returns>A reference to the <see cref="IResourceBuilder{T}"/>.</returns>
-    public static IResourceBuilder<AzureBicepServiceBusResource> AddBicepAzureServiceBus(this IDistributedApplicationBuilder builder, string name, string[]? queueNames = null, string[]? topicNames = null)
+    public static IResourceBuilder<AzureBicepServiceBusResource> AddBicepAzureServiceBus(this IDistributedApplicationBuilder builder, string name)
     {
         var resource = new AzureBicepServiceBusResource(name);
-        // TODO: Change topics and queues to child resources
 
         return builder.AddResource(resource)
                       .WithParameter("serviceBusNamespaceName", resource.CreateBicepResourceName())
                       .WithParameter(AzureBicepResource.KnownParameters.PrincipalId)
                       .WithParameter(AzureBicepResource.KnownParameters.PrincipalType)
-                      .WithParameter("topics", topicNames ?? [])
-                      .WithParameter("queues", queueNames ?? [])
+                      // TODO: Shouldn't these be lazily evaluated?
+                      .WithParameter("queues", resource.Queues)
```

Comment:
We need to find a nice way to make these lazily evaluated.

Reply:
@mitchdenny just added a callback API